src/app/reporting/Engine.cpp

```diff
@@ -492,10 +492,11 @@ CHIP_ERROR Engine::BuildAndSendSingleReportData(ReadHandler * apReadHandler)
     CHIP_ERROR err = CHIP_NO_ERROR;
     chip::System::PacketBufferTLVWriter reportDataWriter;
     ReportDataMessage::Builder reportDataBuilder;
-    chip::System::PacketBufferHandle bufHandle = System::PacketBufferHandle::New(chip::app::kMaxSecureSduLengthBytes);
+    chip::System::PacketBufferHandle bufHandle = nullptr;
     uint16_t reservedSize                      = 0;
     bool hasMoreChunks                         = false;
     bool needCloseReadHandler                  = false;
+    size_t maxSduSize                          = 0;
```

Comment:
What is an "Sdu"

This should be something like `reportBufferMaxSize` or something.

Reply:
Will rename. I went with how the constant was defined in StatusResponse.h. I guess SDU stands for Service Data Unit, as is the canonical defintion, similar to PDU etc. Not sure.
This definition in StatusResponse.h also seems redundant and is essentially the same as one of the constants in MessageHeader.h. I felt that we probably need to consolidate the definitions into a single place(at least not leave them strewn around in various locations) but did not want to do that in this PR.
StatusResponse.h does not seem like the ideal place for this definition too, isn't it?